compactor/src/compact.rs

```diff
@@ -1386,6 +1663,266 @@ mod tests {
+        assert_eq!(count, 4);
+
+        // create tombstones
+        // ts1 overlaps with pf1 and pf2 but issued before pf1 and pf2 hence it won't be used
+        let ts1 = table
+            .with_sequencer(&sequencer)
+            .create_tombstone(2, 6000, 21000, "tag1=UT")
+            .await;
+        // ts2 overlaps with both pf3 and pf2
+        let ts2 = table
+            .with_sequencer(&sequencer)
+            .create_tombstone(11, 6000, 12000, "tag1=VT")
+            .await;
+        // ts3 does not overlap with any files
+        let ts3 = table
+            .with_sequencer(&sequencer)
+            .create_tombstone(22, 1000, 2000, "tag1=VT")
+            .await;
+        // ts4 overlpas with pf4
+        let ts4 = table
+            .with_sequencer(&sequencer)
+            .create_tombstone(24, 27000, 28000, "tag2=AA")
+            .await;
+        // should have 4 tomstones
+        let count = catalog.count_tombstones_for_table(table.table.id).await;
+        assert_eq!(count, 4);
+        // should not have any processed tombstones for any tombstones
+        let count = catalog.count_processed_tombstones(ts1.tombstone.id).await;
+        assert_eq!(count, 0);
+        let count = catalog.count_processed_tombstones(ts2.tombstone.id).await;
+        assert_eq!(count, 0);
+        let count = catalog.count_processed_tombstones(ts3.tombstone.id).await;
+        assert_eq!(count, 0);
+        let count = catalog.count_processed_tombstones(ts4.tombstone.id).await;
+        assert_eq!(count, 0);
+
+        // -------------------------------------------------------------------------------
+        // Step 2: Get cadidate partitions and their files with tombstones to be compacted
+
+        // Get files that need to apply the available tombstones
+        let partitions_with_files = compactor.partitions_with_files_to_compact().await.unwrap();
+        // should inlcude only one partition
+        assert_eq!(partitions_with_files.len(), 1);
+
+        // Verify the its files
+        let files = partitions_with_files[0].files_with_tombstones.clone();
+        // should have 3 files
+        assert_eq!(files.len(), 3);
+        // Expected parquet files with tombstones
+        let pfwts2 = Arc::new(ParquetFileWithTombstone {
+            data: Arc::new(pf2),
+            tombstones: vec![ts2.tombstone.clone()],
+        });
+        let pfwts3 = Arc::new(ParquetFileWithTombstone {
+            data: Arc::new(pf3),
+            tombstones: vec![ts2.tombstone.clone()],
+        });
+        let pfwts4 = Arc::new(ParquetFileWithTombstone {
+            data: Arc::new(pf4),
+            tombstones: vec![ts4.tombstone.clone()],
+        });
+        assert!(files.contains(&pfwts2));
+        assert!(files.contains(&pfwts3));
+        assert!(files.contains(&pfwts4));
+
+        // cast to FileMeta
+        let files = files
+            .iter()
+            .map(|f| Arc::clone(&(*f)) as Arc<dyn FileMeta>)
```

Comment:
This can be a bit simpler:

```suggestion
            .map(|f| Arc::clone(f) as Arc<dyn FileMeta>)
```